### PR TITLE
Utilise Resource configmap

### PR DESCRIPTION
**Pull request by tomahawk28**
- `app` CR controller is out of scope, it should create `configmap` inside tenant cluster directly, so we should careful

### CODE REVIEW

File: service/controller/appcatalog/v1/resource_set.go [hunk 1]
```diff
@@ -56,7 +57,20 @@ func NewResourceSet(config ResourceSetConfig) (*controller.ResourceSet, error) {
...
+		stateGetter, err := index.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		configOps := configmap.Config{
+			K8sClient: config.K8sClient,
+			Logger:    config.Logger,
+
+			Name:        config.ProjectName,
```
Comment (kopiczko) [inline service/controller/appcatalog/v1/resource_set.go]: The name here should be "configmap". This is what `resource.Name()` returns.

File: service/controller/app/v1/resource_set.go [hunk 2]
```diff
@@ -155,11 +155,13 @@ func NewResourceSet(config ResourceSetConfig) (*controller.ResourceSet, error) {
 			return nil, microerror.Mask(err)
 		}
 
-		// TODO: Remove if statement after in-cluster flag is implemented.
 		var k8sClient kubernetes.Interface
 		var g8sClient versioned.Interface
 
-		if cr.Spec.KubeConfig.Secret.Name != "" {
+		if key.InCluster(cr) {
```
Comment (kopiczko) [inline service/controller/app/v1/resource_set.go]: Can `InCluster` related changes land in a separate PR?

File: service/controller/appcatalog/v1/resource_set.go (hunk 1 above)
Comment (tomahawk28) [reply service/controller/appcatalog/v1/resource_set.go]: I will make it constant

File: service/controller/app/v1/resource_set.go (hunk 2 above)
Comment (tomahawk28) [reply service/controller/app/v1/resource_set.go]: No, Since it's already in master after your initial review, it's gone now.

File: service/controller/appcatalog/v1/resource_set.go (hunk 1 above)
Comment (kopiczko) [reply service/controller/appcatalog/v1/resource_set.go]: Actually you can do that:

```suggestion
			Name:        index.Name,
```

File: service/controller/app/v1/resource_set.go (hunk 2 above)
Comment (kopiczko) [reply service/controller/app/v1/resource_set.go]: Sorry I don't follow. You mean it will disappear after merging master branch?

Comment (tomahawk28) [reply service/controller/app/v1/resource_set.go]: My bad; I just rebase it with master and it's gone :)

File: service/controller/appcatalog/v1/resource_set.go (hunk 1 above)
Comment (tomahawk28) [reply service/controller/appcatalog/v1/resource_set.go]: is it okay to have `indexv1`?

File: service/controller/appcatalog/v1/resource/index/current_test.go [hunk 3]
```diff
@@ -122,13 +122,15 @@ func Test_Resource_GetCurrentState(t *testing.T) {
 				t.Fatal("expected non-nil ConfigMap got nil")
 			}
 
-			cm, err := toConfigMap(result)
-			if err != nil {
-				t.Fatalf("error == %#v, want nil", err)
-			}
+			if result != nil {
```
Comment (tomahawk28) [inline service/controller/appcatalog/v1/resource/index/current_test.go]: if not check this condition, then we are falling into memory error.

**Comment by kopiczko** [comment]
Nice. LGTM. Let's test that once more, merge operatorkit changes, vendor it again and merge. Can you open that PR to public?

File: service/controller/appcatalog/v1/resource_set.go (hunk 1 above)
Comment (kopiczko) [reply service/controller/appcatalog/v1/resource_set.go]: Yes it's fine.

File: service/controller/appcatalog/v1/resource/index/current_test.go (hunk 3 above)
Comment (kopiczko) [reply service/controller/appcatalog/v1/resource/index/current_test.go]: I think we should check if `len(result) == 1`.

Comment (tomahawk28) [reply service/controller/appcatalog/v1/resource/index/current_test.go]: is it okay when `result` is nil?

Comment (kopiczko) [reply service/controller/appcatalog/v1/resource/index/current_test.go]: Yes.

Comment (kopiczko) [reply service/controller/appcatalog/v1/resource/index/current_test.go]: https://play.golang.org/p/X9R3OmCAxcm
